compiler/src/compile.re

```diff
@@ -267,5 +291,13 @@ let () =
 let () =
   Env.compile_module_dependency :=
     (
-      (input, outfile) => ignore(compile_file(~outfile, ~reset=false, input))
+      (input, outfile) =>
+        ignore(
+          compile_file(
+            ~outfile,
+            ~reset=false,
+            ~hook=stop_after_object_file_emitted,
```

Comment:
Previously, it would just compile completely, but now since linking is a step that also happens, this skips linking when compiling dependencies (since they'll all be linked together at the end).